src/history.rs

```diff
@@ -475,8 +476,12 @@ impl HistoryImpl {
 
         let _profiler = TimeProfiler::new("load_old");
         if let Some(filename) = history_filename(&self.name, L!("")) {
-            let Ok(fd) = wopen_cloexec(&filename, OFlag::O_RDONLY, Mode::empty()) else {
-                return;
+            let mut file = {
+                let Ok(fd) = wopen_cloexec(&filename, OFlag::O_RDONLY, Mode::empty()) else {
+                    return;
+                };
+                // TODO: Return `File` directly from wopen_cloexec
+                unsafe { File::from_raw_fd(fd.into_raw_fd()) }
```

Comment:
Why not safe from? That's kinda the main reason why I made it an `OwnedFd`.
```suggestion
                File::from(fd)
```

Reply:
Thanks. I forgot about that conversion!